index.js

```diff
@@ -141,44 +184,64 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let ticketNum = 0
+
+function takeATicketNumber(line) {
+  ticketNum++
+  line.push(ticketNum)
+  return `Welcome. You are ticket number ${ticketNum}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc(){
+  let ticketNum = 0
+
+  function takeATicketNumber(line) {
+    ticketNum++
+    line.push(ticketNum)
+    return `Welcome. You are ticket number ${ticketNum}`
+  }
+
+  return takeATicketNumber
+}
```

Comment:
This is great. We can refactor by removing line 227 and adding the return in front of 221